src/all/cosplaytele/src/eu/kanade/tachiyomi/extension/all/cosplaytele/CosplayTele.kt

```diff
@@ -0,0 +1,227 @@
+    override fun popularMangaSelector(): String = ""
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        launchIO { fetchFilters() }
+        val jsonObject = json.decodeFromString<JsonArray>(response.body.string())
+        val mangas = jsonObject.map { item ->
+            val head = item.jsonObject["yoast_head_json"]!!.jsonObject
+            val manga = SManga.create().apply {
+                title = head["og_title"]!!.jsonPrimitive.content
+                thumbnail_url = head["og_image"]!!.jsonArray[0].jsonObject["url"]!!.jsonPrimitive.content
+            }
+            manga.setUrlWithoutDomain(head["og_url"]!!.jsonPrimitive.content)
+            manga
+        }
+        return MangasPage(mangas, mangas.size >= popularPageLimit)
+    }
+
+    // Search
+    override fun searchMangaFromElement(element: Element) = latestUpdatesFromElement(element)
+    override fun searchMangaNextPageSelector() = latestUpdatesNextPageSelector()
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val filterList = if (filters.isEmpty()) getFilterList() else filters
+        val categoryFilter = filterList.findInstance<UriPartFilter>()!!
+        return when {
+            query.isEmpty() && categoryFilter.state != 0 -> GET("$baseUrl/${categoryFilter.toUriPart()}/page/$page/")
+            query.isNotEmpty() && categoryFilter.state != 0 -> GET("$baseUrl/${categoryFilter.toUriPart()}/page/$page/?s=$query")
+            query.isNotEmpty() -> GET("$baseUrl/page/$page/?s=$query")
+            else -> latestUpdatesRequest(page)
+        }
+    }
+
+    override fun searchMangaSelector() = latestUpdatesSelector()
+
+    // Details
+    override fun mangaDetailsParse(document: Document): SManga {
+        val manga = SManga.create()
+        manga.title = document.select(".entry-title").text()
+        manga.description = document.select(".entry-title").text()
+        val genres = getTags(document)
+        manga.genre = genres.joinToString(", ")
+        manga.status = SManga.COMPLETED
+        return manga
+    }
+
+    private fun getTags(document: Element): List<String> {
+        val pattern = """.*/(tag|category)/.*""".toRegex()
+        return document.select("#main a").filter { a -> pattern.matches(a.attr("href")) }.map { a ->
+            val link = a.attr("href").split(".com/")[1]
+            val tag = a.text()
+            Log.d("app.mihon.debug", "here $tag/$link")
+            if (tag.isNotEmpty()) {
+                this.categories[tag] = link
+            }
+            tag
+        }
+    }
+
+    override fun chapterFromElement(element: Element): SChapter {
+        val chapter = SChapter.create()
+        chapter.setUrlWithoutDomain(element.select("link[rel=\"canonical\"]").attr("href"))
+        chapter.chapter_number = -2f
```

Comment:
```suggestion
```
why -2f?

Reply:
That's what I saw in Everia.Club (I just left it as is)